packages/react-ui-kit/src/Form/Tooltip.tsx

```diff
@@ -17,135 +17,146 @@
-    left: '100%',
-    marginLeft: '8px !important',
-    top: 0,
-    bottom: 0,
-    margin: 'auto 0',
-    '& .tooltip-arrow': {
-      filter: 'drop-shadow(-2px 0px 1px rgba(0, 0, 0, 0.1))',
-      borderTop: '8px solid transparent',
-      borderBottom: '8px solid transparent',
-      borderRight: `8px solid ${theme.Tooltip.backgroundColor}`,
-      top: '50%',
-      transform: 'translateY(-50%)',
-      left: -7,
+    borderLeft: '6px solid transparent',
+    borderRight: '6px solid transparent',
+    position: 'absolute',
+
+    "&[data-position='top']": {
+      borderTop: `10px solid ${theme.Tooltip.backgroundColor}`,
+      bottom: '-6px',
     },
-  },
-  "&[data-position='bottom'] .tooltip-content": {
-    top: '100%',
-    marginTop: 10,
-    left: '50%',
-    transform: 'translateX(-50%)',
-    '& .tooltip-arrow': {
-      filter: 'drop-shadow(0px -2px 1px rgba(0, 0, 0, 0.1))',
-      borderLeft: '12px solid transparent',
-      borderRight: '12px solid transparent',
-      borderBottom: `12px solid ${theme.Tooltip.backgroundColor}`,
-      bottom: '90%',
-      left: '50%',
-      transform: 'translateX(-50%)',
-    },
-  },
-  "&[data-position='left'] .tooltip-content": {
-    right: '100%',
-    marginRight: '8px !important',
-    top: 0,
-    bottom: 0,
-    margin: 'auto 0',
-    '& .tooltip-arrow': {
-      filter: 'drop-shadow(2px 0px 1px rgba(0, 0, 0, 0.1))',
-      borderTop: '8px solid transparent',
-      borderBottom: '8px solid transparent',
-      borderLeft: `8px solid ${theme.Tooltip.backgroundColor}`,
-      top: '50%',
-      transform: 'translateY(-50%)',
-      right: -7,
+    "&[data-position='bottom']": {
+      borderBottom: `10px solid ${theme.Tooltip.backgroundColor}`,
+      top: '-6px',
     },
   },
 });
 
-const filterTooltipProps = (props: ToolTipProps) => filterProps(props, ['position', 'body', 'isOpen']);
+interface PortalProps {
+  children: ReactNode;
+  bounding?: DOMRect;
+}
+
+const PortalComponent = ({children, bounding}: PortalProps) => {
+  const bodyElement = document.querySelector('#wire-app');
+
+  const [isTouchingTop, setIsTouchingTop] = useState(false);
 
-export const Tooltip = ({children, ...props}: ToolTipProps) => {
+  if (!bodyElement) {
+    return null;
+  }
+
+  return createPortal(
+    <div
+      ref={element => {
+        if (!bounding) {
+          return;
+        }
+
+        if (element) {
+          const isTouchingTopEdge = bounding.y <= element.clientHeight + paddingTop * 2;
+          setIsTouchingTop(isTouchingTopEdge);
+
+          const elementWidth = (element.scrollWidth - bounding.width) / 2;
+          element.style.left = `${bounding.x - elementWidth}px`;
+
+          if (isTouchingTopEdge) {
+            element.style.top = `${bounding.y + bounding.height + paddingTop}px`;
+          } else {
+            element.style.top = `${bounding.y - element.clientHeight - paddingTop}px`;
+          }
+        }
+      }}
```

Comment:
**suggestion**: I think this would be much more readable if it lived above the return statement.